sqldelight-core/src/test/query-interface-fixtures/query-requires-type/com/sample/JoinedWithUsing.kt

```diff
@@ -4,19 +4,21 @@ import kotlin.Boolean
 import kotlin.String
 
 interface JoinedWithUsing {
-  fun name(): String
+  val name: String
 
-  fun is_cool(): Boolean
+  val is_cool: Boolean
 
-  data class Impl(override val name: String, override val is_cool: Boolean) : JoinedWithUsingKt
+  data class Impl(override val name: String, override val is_cool: Boolean) : JoinedWithUsing
 }
 
-interface JoinedWithUsingKt : JoinedWithUsing {
-  val name: String
+abstract class JoinedWithUsingModel : JoinedWithUsing {
+  override val name: String
+    get() = name()
 
-  val is_cool: Boolean
+  override val is_cool: Boolean
```

Comment:
I can't remember. Are these final or do we need to mark them explicitly final?

Reply:
good call they need to be explicit